backend/app/summary.py

```diff
@@ -253,6 +251,39 @@ def update(sqldb) -> None:
         }
     )
 
+    # collect all app IDs to update
+    apps_to_update = {}
+    for app_id, data in summary_dict.items():
+        try:
+            summary_json = json.loads(json.dumps(data, cls=JSONSetEncoder))
+            apps_to_update[app_id] = summary_json
+        except Exception as e:
+            print(f"Error encoding summary data for {app_id}: {str(e)}")
+            continue
+
+    # update all apps in a single transaction
+    try:
+        for app_id, summary_json in apps_to_update.items():
+            app = models.App.by_appid(sqldb, app_id)
+            if app:
+                appstream_data = app.appstream
```

Comment:
Do we really need to write the appstream data here too? It should not change here, correct? Can't we just touch summaries and be done?